sql/2021/privacy/most_common_featurepolicy_permissionspolicy_directive_values.sql

```diff
@@ -0,0 +1,141 @@
+#standardSQL
+# Most common directives for Feature-Policy or Permissions-Policy
+
+WITH page_ranks AS (
+  SELECT
+    client,
+    page,
+    rank
+  FROM
+    `httparchive.almanac.summary_response_bodies`
+  WHERE
+    date = '2021-07-01' AND firstHtml = TRUE
+),
+
+response_headers AS (
+  SELECT
+    client,
+    page,
+    LOWER(JSON_VALUE(response_header, '$.name')) AS header_name,
+    LOWER(JSON_VALUE(response_header, '$.value')) AS header_value
+  FROM
+    `httparchive.almanac.summary_response_bodies`,
```

Comment:
Why do we need to use `response_bodies`? Try our best not to use that is at al possible. Can we use `summary_requests` (which is actually requests and responses btw):

```suggestion
    `httparchive.almanac.summary_requests`,
```

Reply:
Is this supposed to be a drop-in replacement? I don't see a `response_headers` column in `summary_requests`.

Reply:
Sorry my mistake! It’s the `requests` table not `summary_requests`

Reply:
Does this actually matter? We'll be accessing the same columns anyway, so it should be exactly the same data (size).
Unless it's about the 50 million extra rows in `requests` versus `summary_response_bodies`?
(obviously it's just a search/replace, so not much effort to change, but I'm curious as to the reason why)

Reply:
You're not wrong in this case.

But I still think we should change it, for a few reasons:

The `response_bodies` table just scares the heck outta me because it's so massive! It costs $227 to query the almanac version in full, and $162 (per month!) for the monthly version!

It's a red flag to me when I see it used. This year we scanned our 2020 code base for any use of it to see if we could implement those metrics for cheaper in custom metrics. Having it used like this will just add noise to that.

Yes the `almanac` version of that table (`almanac.summary_response_bodies`) is partitioned on certain key columns so you can just be charged for those columns but it's still easy to get the full cost, for example when:
- Changing to `SELECT *` for debugging
- Changing to monthly tables to see if it's changed since the Almanac was published (we make the queries easily available on the Web Almanac so try to assume someone might run them themselves, even if no one does).

Plus it's a simple drop in replacement - no further code changes required! So no real reason NOT to change it IMHO.

Does that make sense?

Reply:
Sure; fixed in ec2635c.